packages/import/src/importIssues.ts

```diff
@@ -385,3 +360,116 @@ const createIssueWithRetries = async (
     }
   }
 };
+
+const handleLabels = async (
+  client: LinearClient,
+  data: ImportResult,
+  teamId: string,
+  state: {
+    existingLabelMap: { [name: string]: string };
+    existingLabelGroupsMap: { [name: string]: string };
+    existingGroupIdLabelMap: { [id: Id]: { [name: string]: Id } };
+  }
+) => {
+  const { existingLabelMap, existingLabelGroupsMap, existingGroupIdLabelMap } = state;
+  const { labels } = data;
+
+  const createLabel = async ({
+    name,
+    description,
+    color,
+    parentId,
+  }: {
+    name: string;
+    description?: string;
+    color?: string;
+    parentId?: string;
+  }) => {
+    const labelResponse = await client.createIssueLabel({
+      name,
+      description,
+      color,
+      teamId,
+      parentId,
+    });
+
+    const issueLabel = await labelResponse?.issueLabel;
+    return issueLabel?.id;
+  };
+
+  // Create labels and mapping to source data
+  const labelMapping = {} as { [id: string]: Id };
+
+  for (const labelId of Object.keys(labels)) {
+    const label = labels[labelId];
+    let labelName = _.truncate(label.name.trim(), { length: 80 });
+
+    let groupLabelId: string | undefined;
+    // Handle label groups
+    if (labelName.indexOf("/") !== -1) {
+      const parts = labelName.split("/");
+      const group = parts.slice(0, -1).join("/");
+      const subgroup = parts[parts.length - 1];
+
+      groupLabelId = existingLabelGroupsMap[group.toLowerCase()];
+
+      const rootLabelExists = existingLabelMap[group.toLowerCase()] !== undefined;
+
+      // Label group does not exist, create it
+      if (!groupLabelId) {
+        groupLabelId = await createLabel({
+          name: rootLabelExists ? `${group} (group)` : group,
+          color: label.color,
+          description: label.description,
+        });
+
+        if (groupLabelId) {
+          existingLabelGroupsMap[group.toLowerCase()] = groupLabelId;
+          existingGroupIdLabelMap[groupLabelId] = {};
+        }
+      }
+
+      labelName = subgroup; // Use the subgroup name for the actual label
+    }
+
+    let actualLabelId: string | undefined;
+    if (groupLabelId) {
+      // If we have a group label, check if we've already created the subgroup label
+      actualLabelId = existingGroupIdLabelMap[groupLabelId]?.[labelName.toLowerCase()];
+    } else {
+      // Check if this label matches with an existing group label
+      actualLabelId = existingLabelGroupsMap[labelName.toLowerCase()];
+
+      if (actualLabelId) {
+        // This label has matched with an existing group label. We cannot re-use the label as-is, it will be renamed.
+        actualLabelId = undefined;
```

Comment:
Not sure if this is something we want to support, but here we could check if the matching existing group is in the same team or not. We could allow creating groups with the same name as long as they are in different teams.